### ISSUE TITLE
Changes in 13.5.3 break ability for plugins to provide custom changelog prior to version selection

### ISSUE THREAD

**Issue by rwjblue** [release-it/release-it#647]
Prior to the changes in https://github.com/release-it/release-it/commit/6d6f59b8ba4ec6de98a262cecd6337bd2aa26627 a plugin could `this.setContext({ changelog: 'foo' })` and have that changelog be used for the initial prompting of next version. This does not seem to be possible any longer as of 13.5.3+.

6d6f59b8ba4ec6de98a262cecd6337bd2aa26627 refactors the way that the initial changelog is retrieved, from:

https://github.com/release-it/release-it/blob/a0a2a5c8e0a703ce550bce3709a72d1de6a80bcc/lib/tasks.js#L74

And now in 13.5.3+ it is always retrieved from the global context:

https://github.com/release-it/release-it/blob/dda4465bbab1b3225a10bc347b48c57ad3a4f644/lib/tasks.js#L69

As a result this has broken a feature that I added to release-it-lerna-changelog in https://github.com/rwjblue/release-it-lerna-changelog/pull/12.

Suggestions on a path forward?

**Comment by rwjblue** [comment]
Note: I've worked around the issue in https://github.com/rwjblue/release-it-lerna-changelog/pull/38, but this _basically_ feels like a race condition scenario to me and I'd love to come up with a better solution.

**Comment by webpro** [comment]
Oh no, my apologies. I've always wanted to have better tests for plugins. I totally agree it's brittle right now, and this clearly shows it. I'll think about a more robust test suite wrt plugins here.

For now, how about this:

```
const name = await reduceUntil(plugins, plugin => plugin.getName());
const latestVersion = await reduceUntil(plugins, plugin => plugin.getLatestVersion());
const changelog = await reduceUntil(plugins, plugin => plugin.getChangelog());
...
config.setContext({ name, latestVersion, version, changelog });
```

Userland plugins come first, so all plugins will use the changelog from the plugin.

I made a release you can install using `release-it@changelog`, please let me know how this works for you and I'll release it.

**Comment by rwjblue** [comment]
Ya, I like the `getChangelog` hook. That makes sense, and definitely works better than what I did in https://github.com/rwjblue/release-it-lerna-changelog/pull/38 (which relies on effectively unrelated semantic timing of `getLatestVersion` running _just_ before the changelog is retrieved from the global config).

**Comment by webpro** [comment]
Not sure what happened, yesterday I released [v13.5.5](https://github.com/release-it/release-it/releases/tag/13.5.5) with this change, but my comment here got lost. But here it is, so I'm going to close this issue. Let me know if you encounter any issues.

**Comment by rwjblue** [comment]
Thank you very much @webpro!
